vscode-dotnet-runtime-library/src/Acquisition/LinuxVersionResolver.ts

```diff
@@ -0,0 +1,288 @@
+
+/**
+ * An enumeration type representing all distros with their versions that we recognize.
+ * @remarks
+ * Each . in a semver should be represented with _.
+ * The string representation of the enum should contain exactly one space that separates the distro, then the version.
+ */
+export interface DistroVersionPair {
+    distro : string,
+    version : string
+}
+
+/**
+ * @remarks
+ * Distro support means that the distro provides a dotnet sdk package by default without intervention.
+ *
+ * Microsoft support means that Microsoft provides packages for the distro but it's not in the distro maintained feed.
+ * For Microsoft support, we currently don't support installs of these feeds yet.
+ *
+ * Partial support does not have any change in behavior from unsupported currently and can mean whatever the distro maintainer wants.
+ * But it generally means that the distro and microsoft both do not officially support that version of dotnet.
+ *
+ * Unknown is a placeholder for development testing and future potential implementation and should not be used by contributors.
+ */
+export const enum DotnetDistroSupportStatus {
+    Unsupported = 'UNSUPPORTED',
+    Distro = 'DISTRO',
+    Microsoft = 'MICROSOFT',
+    Partial = 'PARTIAL',
+    Unknown = 'UNKNOWN'
+}
+
+/**
+ * This class is responsible for detecting the distro and version of the Linux OS.
+ * It also serves as the entry point to installation via a specific distro implementation
+ * by implementing version validation that normally happens inside of a windows or mac .net installer.
+ * Since those don't exist for linux, we need to manually implement and check certain edge-cases before allowing the installation to occur.
+ */
+export class LinuxVersionResolver
+{
+    private distro : DistroVersionPair | null = null;
+    public distroSDKProvider : IDistroDotnetSDKProvider | null = null;
+    protected commandRunner : ICommandExecutor;
+    protected acquisitionContext : IAcquisitionWorkerContext;
+
+    public conflictingInstallErrorMessage = `A dotnet installation was found which indicates dotnet that was installed via Microsoft package feeds. But for this distro and version, we only acquire .NET via the distro feeds.
+    You should not mix distro feed and microsoft feed installations. To continue, please completely remove this version of dotnet to continue by following https://learn.microsoft.com/dotnet/core/install/remove-runtime-sdk-versions?pivots=os-linux.
+    Your install location: `;
+    public conflictingCustomInstallErrorMessage = `A custom dotnet installation exists.
+    If we were to install another .NET, we would break your custom .NET installation, so the installation request has been refused.
+    If you would like to proceed with installing .NET automatically for VS Code, you must remove this custom installation.
+    Your custom install is located at: `;
+    public baseUnsupportedDistroErrorMessage = 'We are unable to detect the distro or version of your machine';
```

Comment:
These messages are great - I especially love the links to the 'how to un-mess-up-yourself' docs.

Reply:
Thank you, I did put more thought into these :) Glad they are good!